plugin/src/test/java/io/jenkins/plugins/remotingkafka/KafkaKubernetesCloudTest.java

```diff
@@ -21,9 +29,29 @@
     public KubernetesServer k = new KubernetesServer(true, true);
 
     @Test
-    public void testProvision() {
+    public void testProvisionCreateThenTerminatePod() throws Exception {
         KafkaKubernetesCloud cloud = new KafkaKubernetesCloud("kafka-kubernetes");
         cloud.setServerUrl(k.getMockServer().url("/").toString());
+        cloud.setSkipTlsVerify(true);
+        j.jenkins.clouds.add(cloud);
+
+        Collection<NodeProvisioner.PlannedNode> provisionedNodes = cloud.provision(new LabelAtom("test"), 1);
+        assertThat(provisionedNodes, hasSize(1));
+        KafkaCloudSlave slave = (KafkaCloudSlave) provisionedNodes.iterator().next().future.get();
+        TaskListener listener = new LogTaskListener(Logger.getLogger(KafkaKubernetesCloudTest.class.getName()), Level.INFO);
+        PodResource<Pod, DoneablePod> pod = k.getClient().pods().inNamespace(cloud.getNamespace()).withName(slave.getNodeName());
+
+        assertNull(pod.get());
+        j.jenkins.addNode(slave);
+        TimeUnit.SECONDS.sleep(20);
```

Comment:
Hmmm...this seems like a long sleep...could you poll, sleep for 1 second and timeout polling after 30 instead?